### PR TITLE
feat: update runner dockerfile with dependencies and update ci and Readme.

### PR DESCRIPTION
**Pull request by Annavar-satish**
## Check List (Check all the applicable boxes) 

- [X] My code follows the code style of this project.
- [X] My change requires changes to the documentation.
- [X] I have updated the documentation accordingly.
- [X] All new and existing tests passed.
- [X] This PR does not contain plagiarized content.
- [X] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,18 +9,20 @@ D-in-D image for GitHub Actions Self-Hoster runner based on [summerwind/actions-
 | summerwind/actions-runner-dind | v2.311.0-ubuntu-20.04 |
 | Golang | 1.21.5 |
 | Buildx | 0.17.1 |
-| Node | 20.17.0 |
+| Node | 20.18.0 |
```
Comment (francois-codes) [inline README.md:12]: I think our repo's `.tool-versions` mention node 20.17.0, so it may not run on node 20.18.0, but I'm not sure

**Review by francois-codes — CHANGES_REQUESTED** [review]
Thanks @Annavar-satish Please check the node version and wether the build & other scripts are running as expected with the version of node.js you've added.

I think we're also missing turbo & pnpm, which can be installed with npm, but also via brew or other package managers (in fact, maybe it's the npm install -g that  doesn't correctly bind the binary in the runner user path)

File: README.md (hunk 1 above)
Comment (Annavar-satish) [reply README.md:12]: I have verified the `.tool-versions`, but there is no mention of [node version](https://github.com/Cognyx/webapp/blob/dev/.tool-versions) @francois-codes  even though we can control the node version from the workflow file during node setup step.

File: Dockerfile [hunk 2]
```diff
@@ -10,18 +10,26 @@ RUN set -ex; \
...
-  php-redis php-mysqli php-opcache php-pdo php-phar php-posix php-simplexml \
-  php-sockets php-sqlite3 php-tidy php-tokenizer php-xml php-xmlwriter php-zip \
+  php8.2 php8.2-apcu php8.2-bcmath php8.2-dom php8.2-ctype php8.2-curl php8.2-exif php8.2-fileinfo php8.2-fpm \
+  php8.2-gd php8.2-gmp php8.2-iconv php8.2-intl php-json php8.2-mbstring php8.2-mysqlnd php8.2-mysql php8.2-soap \
+  php8.2-redis php8.2-mysqli php8.2-opcache php8.2-pdo php8.2-phar php8.2-posix php8.2-pdo-sqlite php8.2-simplexml \
+  php8.2-sockets php8.2-sqlite3 php8.2-tidy php8.2-tokenizer php8.2-xml php8.2-xmlwriter php8.2-zip php8.2-dev \
   php-pear libgd-tools \
   nodejs \
   git unzip; \
+  npm install -g pnpm wrangler@3.56.0 firebase-tools; \
```
Comment (Annavar-satish) [inline Dockerfile:24]: Hi @francois-codes  Here we are installing the necessary dependencies for the FE. we were running this in the CI before. Please Update me If I am missing any dependencies. and I verified the CI and it is working fine.

Comment (Annavar-satish) [reply Dockerfile:24]: yes confirmed. it's working fine.

**Review by deonvdv — APPROVED** [review]

**Pull request merged**